crates/tabby-common/src/enum_conversion.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::Result;
+
+pub trait DatabaseSerializable: Sized {
+    fn as_db_str(&self) -> &'static str;
+    fn from_db_str(s: &str) -> Result<Self>;
+}
+
+#[macro_export]
```

Comment:
Could also have it generate it using the name of the variant as the string, but I wanted to avoid that in case we have variants that use a different db name for casing or formatting